stormpath/stormpath-sdk-angularjs
# ngRoute support

Closes #47.

This PR adds support for `ngRoute` (in addition to `UI Router`). It does so by exposing a new set of APIs, similar to the ones for `UI Router`.

It has the same feature set and the only difference in the API is the use of the word `route` instead of `state`.
### Notes
- I decided to keep it as separate APIs to not break backward compatibility. The code for this is a bit redundant, but this will be addressed in a later PR, which will break the API.
- When reviewing, see my commits for easier overview
### How to verify
1. Checkout this branch
2. Build using `$ grunt dist --force`
3. Copy files from `dist/` into your Angular app folder
4. Make sure your app is using `ngRoute` instead of `UI Router` (e.g. `example/ng-route-app/`)
5. Verify that routing works at it should by navigating through your routes
6. Try to login and logout and verify that redirection is made correctly
7. Try to access protected routes when logged out and verify that you gets redirected to login
8. Try to access protected routes when logged in and verify that you can see them
9. Verify that the same behaviour works when you load the app at these routes in a new tab
